src/charts/aws-node-termination-handler-2/templates/node.k8s.aws_terminators.yaml

```diff
@@ -0,0 +1,209 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.8.0
+  creationTimestamp: null
+  name: terminators.node.k8s.aws
+spec:
+  group: node.k8s.aws
+  names:
+    kind: Terminator
+    listKind: TerminatorList
+    plural: terminators
+    singular: terminator
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: Terminator is the Schema for the terminators API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: TerminatorSpec defines the desired state of Terminator
+            type: object
+            properties:
+              sqs:
+                description: AWS SQS queue configuration.
+                type: object
+                properties:
+                  attributeNames:
```

Comment:
why is this being exposed? Seems like this should be set somewhere else since NTH is what's processing the messages so setting these doesn't really do anything unless NTH knows how to deal with them.

Reply:
Placing the parameters in the Terminator allow varying parameters for individual queues, and it consolidates the configuration to a single location.  The values are included in the request ([here](https://github.com/aws/aws-node-termination-handler/pull/612/files#diff-6e7d2748740e5e4f87c2b71eeea3d540f23cd38cc39a609354057aaead18b4ebR59-R76)).

Only the QueueURL field is required when creating a Terminator.  All other fields will be assigned a default value if omitted.  Each default value may be configured via the helm chart ([here](https://github.com/aws/aws-node-termination-handler/pull/612/files#diff-d69c26c948786815eaf97e0cc35a7e4fbac48df08ed71dc0d0f7f65420b7ae92R13-R22)).

Reply:
I realize QueueURL is the only required field, but I'm curious if a user would ever need to configure `attributeNames`. It seems to me that they wouldn't. Having the configuration all in one spot is nice for code neatness, but when thinking about TerminatorSpec as the NTH API for the user, it seems odd to have things you can configure that don't have an affect. 

The attributeNames could still be runtime defaulted without being present in the Spec, and if it's interesting what those values are, you could place them in the `Status` of the Terminator to show what runtime defaults were applied without allowing the user to change them.